### ISSUE TITLE
Installation fails on Python 3.7

### ISSUE THREAD

**Issue by bofm** [igorcoding/asynctnt#7]
```console
root@5d91fb6a7515:/app# pip -V
pip 18.0 from /usr/local/lib/python3.7/site-packages/pip (python 3.7)
root@5d91fb6a7515:/app# python -V
Python 3.7.0
root@5d91fb6a7515:/app# pip install asynctnt
Looking in indexes: http://pypi.k.avito.ru/pypi/
Collecting asynctnt
  Downloading https://files.pythonhosted.org/packages/1d/7f/a5be00488bd40ee55d15eae3f0b71db796779b4e24f22a1241c2872f5641/asynctnt-0.1.11.tar.gz (268kB)
    100% |████████████████████████████████| 276kB 2.4MB/s
Collecting PyYAML>=3.12 (from asynctnt)
  Downloading https://files.pythonhosted.org/packages/9e/a3/1d13970c3f36777c583f136c136f804d70f500168edc1edea6daa7200769/PyYAML-3.13.tar.gz (270kB)
    100% |████████████████████████████████| 276kB 4.0MB/s
Building wheels for collected packages: asynctnt, PyYAML
  Running setup.py bdist_wheel for asynctnt ... error
  Complete output from command /usr/local/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-vo45rneu/asynctnt/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-7f0xaqx7 --python-tag cp37:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/asynctnt
  copying asynctnt/instance.py -> build/lib.linux-x86_64-3.7/asynctnt
  copying asynctnt/__init__.py -> build/lib.linux-x86_64-3.7/asynctnt
  copying asynctnt/connection.py -> build/lib.linux-x86_64-3.7/asynctnt
  copying asynctnt/log.py -> build/lib.linux-x86_64-3.7/asynctnt
  copying asynctnt/exceptions.py -> build/lib.linux-x86_64-3.7/asynctnt
  copying asynctnt/_testbase.py -> build/lib.linux-x86_64-3.7/asynctnt
  running egg_info
  writing asynctnt.egg-info/PKG-INFO
  writing dependency_links to asynctnt.egg-info/dependency_links.txt
  writing requirements to asynctnt.egg-info/requires.txt
  writing top-level names to asynctnt.egg-info/top_level.txt
  reading manifest file 'asynctnt.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching '*.h' under directory 'asynctnt'
  writing manifest file 'asynctnt.egg-info/SOURCES.txt'
  creating build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/__init__.py -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/buffer.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/buffer.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/cmsgpuck.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/const.pxi -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/coreproto.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/coreproto.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/db.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/db.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/protocol.c -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/protocol.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/protocol.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/python.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/rbuffer.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/rbuffer.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/request.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/request.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/response.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/response.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/schema.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/schema.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/tnt.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/unicode.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  copying asynctnt/iproto/unicode.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
  running build_ext
  building 'asynctnt.iproto.protocol' extension
  creating build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/asynctnt
  creating build/temp.linux-x86_64-3.7/asynctnt/iproto
  creating build/temp.linux-x86_64-3.7/third_party
  creating build/temp.linux-x86_64-3.7/third_party/msgpuck
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I-Ithird_party -I/usr/local/include/python3.7m -c asynctnt/iproto/protocol.c -o build/temp.linux-x86_64-3.7/asynctnt/iproto/protocol.o
  asynctnt/iproto/protocol.c: In function ‘__Pyx_PyCFunction_FastCall’:
  asynctnt/iproto/protocol.c:35944:12: error: too many arguments to function ‘(struct PyObject * (*)(struct PyObject *, struct PyObject * const*, Py_ssize_t))meth’
       return (*((__Pyx_PyCFunctionFast)meth)) (self, args, nargs, NULL);
              ^
  asynctnt/iproto/protocol.c: In function ‘__Pyx__ExceptionSave’:
  asynctnt/iproto/protocol.c:36254:19: error: ‘PyThreadState’ has no member named ‘exc_type’
       *type = tstate->exc_type;
                     ^
  asynctnt/iproto/protocol.c:36255:20: error: ‘PyThreadState’ has no member named ‘exc_value’
       *value = tstate->exc_value;
                      ^
  asynctnt/iproto/protocol.c:36256:17: error: ‘PyThreadState’ has no member named ‘exc_traceback’
       *tb = tstate->exc_traceback;
                   ^
  asynctnt/iproto/protocol.c: In function ‘__Pyx__ExceptionReset’:
  asynctnt/iproto/protocol.c:36263:22: error: ‘PyThreadState’ has no member named ‘exc_type’
       tmp_type = tstate->exc_type;
                        ^
  asynctnt/iproto/protocol.c:36264:23: error: ‘PyThreadState’ has no member named ‘exc_value’
       tmp_value = tstate->exc_value;
                         ^
  asynctnt/iproto/protocol.c:36265:20: error: ‘PyThreadState’ has no member named ‘exc_traceback’
       tmp_tb = tstate->exc_traceback;
                      ^
  asynctnt/iproto/protocol.c:36266:11: error: ‘PyThreadState’ has no member named ‘exc_type’
       tstate->exc_type = type;
             ^
  asynctnt/iproto/protocol.c:36267:11: error: ‘PyThreadState’ has no member named ‘exc_value’
       tstate->exc_value = value;
             ^
  asynctnt/iproto/protocol.c:36268:11: error: ‘PyThreadState’ has no member named ‘exc_traceback’
       tstate->exc_traceback = tb;
             ^
  asynctnt/iproto/protocol.c: In function ‘__Pyx__GetException’:
  asynctnt/iproto/protocol.c:36323:22: error: ‘PyThreadState’ has no member named ‘exc_type’
       tmp_type = tstate->exc_type;
                        ^
  asynctnt/iproto/protocol.c:36324:23: error: ‘PyThreadState’ has no member named ‘exc_value’
       tmp_value = tstate->exc_value;
                         ^
  asynctnt/iproto/protocol.c:36325:20: error: ‘PyThreadState’ has no member named ‘exc_traceback’
       tmp_tb = tstate->exc_traceback;
                      ^
  asynctnt/iproto/protocol.c:36326:11: error: ‘PyThreadState’ has no member named ‘exc_type’
       tstate->exc_type = local_type;
             ^
  asynctnt/iproto/protocol.c:36327:11: error: ‘PyThreadState’ has no member named ‘exc_value’
       tstate->exc_value = local_value;
             ^
  asynctnt/iproto/protocol.c:36328:11: error: ‘PyThreadState’ has no member named ‘exc_traceback’
       tstate->exc_traceback = local_tb;
             ^
  asynctnt/iproto/protocol.c: In function ‘__Pyx__ExceptionSwap’:
  asynctnt/iproto/protocol.c:36858:22: error: ‘PyThreadState’ has no member named ‘exc_type’
       tmp_type = tstate->exc_type;
                        ^
  asynctnt/iproto/protocol.c:36859:23: error: ‘PyThreadState’ has no member named ‘exc_value’
       tmp_value = tstate->exc_value;
                         ^
  asynctnt/iproto/protocol.c:36860:20: error: ‘PyThreadState’ has no member named ‘exc_traceback’
       tmp_tb = tstate->exc_traceback;
                      ^
  asynctnt/iproto/protocol.c:36861:11: error: ‘PyThreadState’ has no member named ‘exc_type’
       tstate->exc_type = *type;
             ^
  asynctnt/iproto/protocol.c:36862:11: error: ‘PyThreadState’ has no member named ‘exc_value’
       tstate->exc_value = *value;
             ^
  asynctnt/iproto/protocol.c:36863:11: error: ‘PyThreadState’ has no member named ‘exc_traceback’
       tstate->exc_traceback = *tb;
             ^
  asynctnt/iproto/protocol.c: In function ‘__Pyx__ReturnWithStopIteration’:
  asynctnt/iproto/protocol.c:37958:22: error: ‘PyThreadState’ has no member named ‘exc_type’
       if (!__pyx_tstate->exc_type) {
                        ^
  error: command 'gcc' failed with exit status 1

  ----------------------------------------
  Failed building wheel for asynctnt
  Running setup.py clean for asynctnt
  Running setup.py bdist_wheel for PyYAML ... done
  Stored in directory: /root/.cache/pip/wheels/ad/da/0c/74eb680767247273e2cf2723482cb9c924fe70af57c334513f
Successfully built PyYAML
Failed to build asynctnt
Installing collected packages: PyYAML, asynctnt
  Running setup.py install for asynctnt ... error
    Complete output from command /usr/local/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-vo45rneu/asynctnt/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-8437v6y1/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/asynctnt
    copying asynctnt/instance.py -> build/lib.linux-x86_64-3.7/asynctnt
    copying asynctnt/__init__.py -> build/lib.linux-x86_64-3.7/asynctnt
    copying asynctnt/connection.py -> build/lib.linux-x86_64-3.7/asynctnt
    copying asynctnt/log.py -> build/lib.linux-x86_64-3.7/asynctnt
    copying asynctnt/exceptions.py -> build/lib.linux-x86_64-3.7/asynctnt
    copying asynctnt/_testbase.py -> build/lib.linux-x86_64-3.7/asynctnt
    running egg_info
    writing asynctnt.egg-info/PKG-INFO
    writing dependency_links to asynctnt.egg-info/dependency_links.txt
    writing requirements to asynctnt.egg-info/requires.txt
    writing top-level names to asynctnt.egg-info/top_level.txt
    reading manifest file 'asynctnt.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching '*.h' under directory 'asynctnt'
    writing manifest file 'asynctnt.egg-info/SOURCES.txt'
    creating build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/__init__.py -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/buffer.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/buffer.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/cmsgpuck.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/const.pxi -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/coreproto.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/coreproto.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/db.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/db.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/protocol.c -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/protocol.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/protocol.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/python.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/rbuffer.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/rbuffer.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/request.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/request.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/response.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/response.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/schema.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/schema.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/tnt.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/unicode.pxd -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    copying asynctnt/iproto/unicode.pyx -> build/lib.linux-x86_64-3.7/asynctnt/iproto
    running build_ext
    building 'asynctnt.iproto.protocol' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/asynctnt
    creating build/temp.linux-x86_64-3.7/asynctnt/iproto
    creating build/temp.linux-x86_64-3.7/third_party
    creating build/temp.linux-x86_64-3.7/third_party/msgpuck
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I-Ithird_party -I/usr/local/include/python3.7m -c asynctnt/iproto/protocol.c -o build/temp.linux-x86_64-3.7/asynctnt/iproto/protocol.o
    asynctnt/iproto/protocol.c: In function ‘__Pyx_PyCFunction_FastCall’:
    asynctnt/iproto/protocol.c:35944:12: error: too many arguments to function ‘(struct PyObject * (*)(struct PyObject *, struct PyObject * const*, Py_ssize_t))meth’
         return (*((__Pyx_PyCFunctionFast)meth)) (self, args, nargs, NULL);
                ^
    asynctnt/iproto/protocol.c: In function ‘__Pyx__ExceptionSave’:
    asynctnt/iproto/protocol.c:36254:19: error: ‘PyThreadState’ has no member named ‘exc_type’
         *type = tstate->exc_type;
                       ^
    asynctnt/iproto/protocol.c:36255:20: error: ‘PyThreadState’ has no member named ‘exc_value’
         *value = tstate->exc_value;
                        ^
    asynctnt/iproto/protocol.c:36256:17: error: ‘PyThreadState’ has no member named ‘exc_traceback’
         *tb = tstate->exc_traceback;
                     ^
    asynctnt/iproto/protocol.c: In function ‘__Pyx__ExceptionReset’:
    asynctnt/iproto/protocol.c:36263:22: error: ‘PyThreadState’ has no member named ‘exc_type’
         tmp_type = tstate->exc_type;
                          ^
    asynctnt/iproto/protocol.c:36264:23: error: ‘PyThreadState’ has no member named ‘exc_value’
         tmp_value = tstate->exc_value;
                           ^
    asynctnt/iproto/protocol.c:36265:20: error: ‘PyThreadState’ has no member named ‘exc_traceback’
         tmp_tb = tstate->exc_traceback;
                        ^
    asynctnt/iproto/protocol.c:36266:11: error: ‘PyThreadState’ has no member named ‘exc_type’
         tstate->exc_type = type;
               ^
    asynctnt/iproto/protocol.c:36267:11: error: ‘PyThreadState’ has no member named ‘exc_value’
         tstate->exc_value = value;
               ^
    asynctnt/iproto/protocol.c:36268:11: error: ‘PyThreadState’ has no member named ‘exc_traceback’
         tstate->exc_traceback = tb;
               ^
    asynctnt/iproto/protocol.c: In function ‘__Pyx__GetException’:
    asynctnt/iproto/protocol.c:36323:22: error: ‘PyThreadState’ has no member named ‘exc_type’
         tmp_type = tstate->exc_type;
                          ^
    asynctnt/iproto/protocol.c:36324:23: error: ‘PyThreadState’ has no member named ‘exc_value’
         tmp_value = tstate->exc_value;
                           ^
    asynctnt/iproto/protocol.c:36325:20: error: ‘PyThreadState’ has no member named ‘exc_traceback’
         tmp_tb = tstate->exc_traceback;
                        ^
    asynctnt/iproto/protocol.c:36326:11: error: ‘PyThreadState’ has no member named ‘exc_type’
         tstate->exc_type = local_type;
               ^
    asynctnt/iproto/protocol.c:36327:11: error: ‘PyThreadState’ has no member named ‘exc_value’
         tstate->exc_value = local_value;
               ^
    asynctnt/iproto/protocol.c:36328:11: error: ‘PyThreadState’ has no member named ‘exc_traceback’
         tstate->exc_traceback = local_tb;
               ^
    asynctnt/iproto/protocol.c: In function ‘__Pyx__ExceptionSwap’:
    asynctnt/iproto/protocol.c:36858:22: error: ‘PyThreadState’ has no member named ‘exc_type’
         tmp_type = tstate->exc_type;
                          ^
    asynctnt/iproto/protocol.c:36859:23: error: ‘PyThreadState’ has no member named ‘exc_value’
         tmp_value = tstate->exc_value;
                           ^
    asynctnt/iproto/protocol.c:36860:20: error: ‘PyThreadState’ has no member named ‘exc_traceback’
         tmp_tb = tstate->exc_traceback;
                        ^
    asynctnt/iproto/protocol.c:36861:11: error: ‘PyThreadState’ has no member named ‘exc_type’
         tstate->exc_type = *type;
               ^
    asynctnt/iproto/protocol.c:36862:11: error: ‘PyThreadState’ has no member named ‘exc_value’
         tstate->exc_value = *value;
               ^
    asynctnt/iproto/protocol.c:36863:11: error: ‘PyThreadState’ has no member named ‘exc_traceback’
         tstate->exc_traceback = *tb;
               ^
    asynctnt/iproto/protocol.c: In function ‘__Pyx__ReturnWithStopIteration’:
    asynctnt/iproto/protocol.c:37958:22: error: ‘PyThreadState’ has no member named ‘exc_type’
         if (!__pyx_tstate->exc_type) {
                          ^
    error: command 'gcc' failed with exit status 1

    ----------------------------------------
Command "/usr/local/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-vo45rneu/asynctnt/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-8437v6y1/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-install-vo45rneu/asynctnt/
```

**Comment by amppumas** [comment]
It is because of Pyyaml, it was [corrected](https://github.com/tmux-python/tmuxp/issues/399) on version 3.13 and the current version in requirements is 3.12 which has problems with python 3.7

**Comment by bofm** [comment]
I confirm that installation works from master branch.
Note that I had to first install Cython.

I canNOT confirm that the version from master (10786bec6) works.
<script src="https://gist.github.com/bofm/b78818eb1cac0cc9a789e011340f5b8d.js"></script>

**Comment by bofm** [comment]
```python
>>> async def main():
...     c = asynctnt.Connection()
...     try:
...         await c.connect()
...     finally:
...         await c.close()
...
>>> asyncio.run(main())
Connect to Tarantool[127.0.0.1:3301] failed: ConnectionRefusedError(111, "Connect call failed ('127.0.0.1', 3301)"). Retrying in 0.333333 seconds
Connect to Tarantool[127.0.0.1:3301] failed: ConnectionRefusedError(111, "Connect call failed ('127.0.0.1', 3301)"). Retrying in 0.333333 seconds
Connect to Tarantool[127.0.0.1:3301] failed: ConnectionRefusedError(111, "Connect call failed ('127.0.0.1', 3301)"). Retrying in 0.333333 seconds
Connect to Tarantool[127.0.0.1:3301] failed: ConnectionRefusedError(111, "Connect call failed ('127.0.0.1', 3301)"). Retrying in 0.333333 seconds
Connect to Tarantool[127.0.0.1:3301] failed: ConnectionRefusedError(111, "Connect call failed ('127.0.0.1', 3301)"). Retrying in 0.333333 seconds
```

It actually works on 3.7. Sorry. My mistake. Will you release it?

**Comment by igorcoding** [comment]
Fixed in 0.1.12 release
https://github.com/igorcoding/asynctnt/releases/tag/v0.1.12
